static/js/message_store.js

```diff
@@ -304,12 +304,10 @@ exports.update_messages = function update_messages(events) {
 
     // If a topic was edited, we re-render the whole view to get any propagated edits
     // to be updated
-    if (topic_edited) {
-        if (!changed_narrow) {
-            home_msg_list.rerender();
-            if (current_msg_list === message_list.narrowed) {
-                message_list.narrowed.rerender();
-            }
+    if (topic_edited && !changed_narrow) {
+        home_msg_list.rerender();
+        if (current_msg_list === message_list.narrowed) {
+            message_list.narrowed.rerender();
```

Comment:
I think actually we just want to move the `home_msg_list.rerender()` out of the `!changed_narrow()` check.  

In other words, if the narrow changed, we still want to avoid rerendering `message_list.narrowed`, since we just switched narrows earlier in this function (rerendering everything anyway).  The bug here is that we were rerendering the home_msg_list either.